### PR TITLE
Flash: Payment path error reporting & timeouts

### PR DESCRIPTION
**Pull request by AndrejMitrovic**

### CODE REVIEW

File: doc/config.example.yaml [hunk 1]
```diff
@@ -75,6 +75,43 @@ validator:
...
+################################################################################
+flash:
+  # Whether or not the Flash protocol should be supported
+  enabled: true
+  # Timeout for each request in milliseconds
+  timeout: 10000
+  # This is a randomly generated keypair
+  # DO NOT USE THOSE VALUES ANYWHERE
+  # Private seed:    SCT4KKJNYLTQO4TVDPVJQZEONTVVW66YLRWAINWI3FZDY7U4JS4JJEI4
+  # Public address:  GCOQEOHAUFYUAC6G22FJ3GZRNLGVCCLESEJ2AXBIJ5BJNUVTAERPLRIJ
```
Comment (Geod24) [inline doc/config.example.yaml:91]: Outdated

Comment (AndrejMitrovic) [reply doc/config.example.yaml:91]: Then it means the rest of the documented config is also outdated.

File: source/agora/flash/ErrorCode.d [hunk 2]
```diff
@@ -19,6 +19,9 @@ public enum ErrorCode : ushort
     /// No error.
     None = 0,
 
+    /// Couldn't find a path to another node, or it was too long
+    PathNotFound,
+
```
Comment (Geod24) [inline source/agora/flash/ErrorCode.d]: Perhaps you want to keep `Unknown` next to `None` ?

File: doc/config.example.yaml (hunk 1 above)
Comment (Geod24) [reply doc/config.example.yaml:91]: True, let's just fix it all at once

**Review by Geod24 — APPROVED** [review]
LGTM, merge at will. One thing for the future is that we should probably be smarter about the config, since flash and validator are quite overlapping.

**Review by omerfirmak — APPROVED** [review]

**Pull request merged**